Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs

```diff
@@ -175,6 +175,289 @@ public void TestExecuteAsyncWithMaxRetryReached()
+                {
+                    // Arrange
+                    cmd.CommandText = $"select 1;";
+
+                    // Act
+                    var row = cmd.ExecuteScalar();
+
+                    // Assert
+                    Assert.AreEqual(1, row);
+                }
+
+                conn.Close();
+            }
+
+            // Get results of the async exec query
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                await conn.OpenAsync(CancellationToken.None).ConfigureAwait(false);
+
+                using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand())
+                {
+                    // Act
+                    var reader = await cmd.GetResultsFromQueryIdAsync(queryId, CancellationToken.None).ConfigureAwait(false);
+                    var queryStatus = await cmd.GetQueryStatusAsync(queryId, CancellationToken.None).ConfigureAwait(false);
+
+                    // Assert
+                    Assert.IsTrue(reader.Read());
+                    Assert.AreEqual($"waited {expectedWaitTime} seconds", reader.GetString(0));
+                    Assert.AreEqual(QueryStatus.Success, queryStatus);
+                }
+
+                await conn.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+            }
+        }
+
+        [Test]
+        public async Task TestAsyncExecCancelWhileGettingResultsAsync()
+        {
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                await conn.OpenAsync(CancellationToken.None).ConfigureAwait(false);
+
+                using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand())
+                {
+                    // Arrange
+                    CancellationTokenSource cancelToken = new CancellationTokenSource();
+                    cmd.CommandText = $"CALL SYSTEM$WAIT(60, \'SECONDS\');";
+
+                    // Act
+                    var queryId = await cmd.ExecuteAsyncInAsyncMode(CancellationToken.None).ConfigureAwait(false);
+                    var queryStatus = await cmd.GetQueryStatusAsync(queryId, CancellationToken.None).ConfigureAwait(false);
+
+                    // Assert
+                    Assert.IsTrue(QueryStatuses.IsStillRunning(queryStatus));
+
+                    // Act
+                    cancelToken.Cancel();
+#if NET471
+                    var thrown = Assert.ThrowsAsync<OperationCanceledException>(async () =>
+                        await cmd.GetResultsFromQueryIdAsync(queryId, cancelToken.Token).ConfigureAwait(false));
+#else
+                    var thrown = Assert.ThrowsAsync<TaskCanceledException>(async () =>
+                        await cmd.GetResultsFromQueryIdAsync(queryId, cancelToken.Token).ConfigureAwait(false));
+#endif
+
+                    // Assert
+#if NET472
```

Comment:
what about NET471?
what about targeting different dotnet framewrks (I'm thinking about .netstandard 2.0)?

Reply:
NET471 has a `The operation was canceled` message covered in the else statement

I'll check the other NET versions

Reply:
This is now fixed because the check for a cancelled token is re-added. Tested on all netstandard 2.0 supported versions and confirmed they all throw the same error